lib/timer.py

```diff
@@ -22,69 +21,88 @@
 # Store the default run_interval_seconds for each job in this dict.
 # OrderedDict is used to ensure each job is run sequentially in the order
 # listed as some jobs may depend on completion of the previous job.
-UACLIENT_JOBS = collections.OrderedDict(
-    [
-        (
-            "update_messaging",
-            {
-                "job_func": update_apt_and_motd_messages,
-                "run_interval_seconds": UPDATE_MESSAGING_INTERVAL,
-            },
-        ),
-        (
-            "update_status",
-            {
-                "job_func": update_status,
-                "run_interval_seconds": UPDATE_STATUS_INTERVAL,
-            },
-        ),
-    ]
-)  # type: Dict[str, Dict[str, Any]]
-
-
-def run_job(
-    job_name: str, job_func: "Callable[[UAConfig], None]", cfg: UAConfig
-) -> bool:
-    """Run a job in a failsafe manner, returning True on success.
-
-    :param job_name: Name of the job to run
-    :param job_func: Job callable to call
-    :param cfg: UAConfig instance
-
-    :return: A bool True when successfully run. False if ignored or in error.
-    """
-    LOG.debug("Running job: %s", job_name)
-    try:
-        job_func(cfg)
-    except Exception as e:
-        LOG.warning("Error executing job %s: %s", job_name, str(e))
-        return False
 
-    return True
+
+class TimedJob:
+    def __init__(
+        self,
+        name: str,
+        job_func: Callable[[UAConfig], None],
+        default_interval_seconds: int,
+    ):
+        self.name = name
+        self._job_func = job_func
+        self._default_interval_seconds = default_interval_seconds
+
+    def run(self, cfg: UAConfig):
+        """Run a job in a failsafe manner, returning True on success.
+        Checks if the job is not disabled before running it.
+
+        :param cfg: UAConfig instance
+
+        :return: A bool True when successfully run. False if ignored
+            or in error.
+        """
+        if not self._should_run(cfg):
+            return False
+
+        LOG.debug("Running job: %s", self.name)
+        try:
+            self._job_func(cfg)
+        except Exception as e:
+            LOG.warning("Error executing job %s: %s", self.name, str(e))
+            return False
+
+        return True
+
+    def run_interval_seconds(self, cfg: UAConfig):
+        """Return the run_interval for the job based on config or defaults."""
+        configured_interval = getattr(cfg, "{}_timer".format(self.name), None)
+        if not isinstance(configured_interval, int) or configured_interval < 0:
+            error_msg = (
+                "Invalid value for the {} interval Found in config. "
```

Comment:
I think we can drop this `the` and replace `Found` for `found`

Reply:
Also, can we test that scenario on the behave test as well ? See if we are logging this part if an invalid config is provided